runtime/src/accounts_db.rs

```diff
@@ -7135,6 +7135,19 @@ impl AccountsDb {
         }
     }
 
+    /// Startup processes can consume large amounts of memory while inserting accounts into the index as fast as possible.
+    /// Calling this can slow down the insertion process to allow flushing ot disk to keep pace.
+    fn maybe_throttle_index_generation(&self) {
+        const LIMIT: usize = 10_000_000;
+        while self
+            .accounts_index
+            .get_startup_remaining_items_to_flush_estimate()
+            > LIMIT
+        {
+            sleep(Duration::from_millis(10));
+        }
+    }
```

Comment:
Can you also include some info on how the two constants were chosen? Since these are somewhat magic numbers, I know it would be helpful to me to understand how these numbers came about (and how to know when to change them). That and maybe what systems this was tested on?